File: app/src/main/java/com/example/fasationbottomnavigation/ResizeAnimation.java
package com.example.fasationbottomnavigation;

import android.view.View;
import android.view.animation.Animation;
import android.view.animation.Transformation;

public class ResizeAnimation extends Animation {

    View view;

    final int startWidth;
    final int targetWidth;

    final int startHeight;
    final int targetHeight;

    public ResizeAnimation(View view, int targetWidth, int targetHeight) {
        this.view = view;

        this.startWidth = view.getWidth();
        this.startHeight = view.getHeight();

        this.targetWidth = targetWidth;
        this.targetHeight = targetHeight;
    }

    @Override
    protected void applyTransformation(float interpolatedTime, Transformation t) {

        int newWidth = (int) (startWidth + (targetWidth - startWidth) * interpolatedTime);
        int newHeight = (int) (startHeight + (targetHeight - startHeight) * interpolatedTime);

        view.getLayoutParams().width = newWidth;
        view.getLayoutParams().height = newHeight;

        view.requestLayout();
    }

    @Override
    public void initialize(int width, int height, int parentWidth, int parentHeight) {
        super.initialize(width, height, parentWidth, parentHeight);
    }

    @Override
    public boolean willChangeBounds() {
        return true;
    }
}

File: app/src/main/java/fasation/bottom/navigation/ResizeAnimation.kt
package fasation.bottom.navigation

import android.view.View
import android.view.animation.Animation
import android.view.animation.Transformation

class ResizeAnimation internal constructor(private var view: View, private val targetWidth: Int, private val targetHeight: Int) : Animation() {

    private val startWidth: Int = view.width
    private val startHeight: Int = view.height

    override fun applyTransformation(interpolatedTime: Float, t: Transformation) {

        val newWidth = (startWidth + (targetWidth - startWidth) * interpolatedTime).toInt()
        val newHeight = (startHeight + (targetHeight - startHeight) * interpolatedTime).toInt()

        view.layoutParams.width = newWidth
        view.layoutParams.height = newHeight

        view.requestLayout()
    }

    override fun willChangeBounds(): Boolean {
        return true
    }
}
